### PR TITLE
Adding a VMware Foreman Sequence Image to clarify

### PR DESCRIPTION
**Pull request by ckyriakidou**
Adding clarifying sequence image to make this easier to consume by end users.

### CODE REVIEW

**Comment by ekohl** [comment]
I always like diagrams because they make it easier to understand. I do wonder how you made it and if there's some source file. Having that in git makes it easier to fix a mistake in the future. Also wondering about SVG instead of PNG, but don't know if this actually renders.

@lzap thoughts? Might be good to have some info about this in the repo's README

**Comment by lzap** [comment]
First off, thanks for your contribution. We need more of these as some things in Foreman are really complex!

We do have couple of diagrams available in our old documentation, we created them using an online tool https://sequencediagram.org - would you mind drawing it too in this tool? It's actually a text base input, similarly to dot/graphviz. We still have the source files as they were committed to git:

https://github.com/theforeman/theforeman.org/tree/gh-pages/static/images/diagrams

@melcorr @spetrosi is it a good time to actually take those and incorporate them into our docs? I think we should break them, provisioning should be one diagram and then we would add Puppet and Ansible separately.

**Comment by ckyriakidou** [comment]
@lzap @ekohl I've changed it using the sequence diagram and resubmitted. Let me know if that still makes sense. (any PR on the actual content would be hugely appreciated).

**Comment by melcorr** [comment]
@ckyriakidou  euxaristw poly!

File: guides/doc-Provisioning_Guide/images/ProvisioningUserData-Sequence.txt [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+participant User
+participant Satellite
+participant VCenter
+participant Capsule X
+
+participant VM
+
+User --> Satellite:Provision VM in network X
+Satellite --> VCenter:Create a VM from template (port 443)
+VCenter --> VM: Clones VM from template
```
Comment (spetrosi) [inline guides/doc-Provisioning_Guide/images/ProvisioningUserData-Sequence.txt:11]:
```suggestion
VCenter --> VM: Clone VM from template
```

File: guides/doc-Provisioning_Guide/images/ProvisioningUserData-Sequence.txt [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+VM --> Capsule X: Call Back to Sat port 80 via Capsule X 8000
+Capsule X --> Satellite: forward Userdata on 80
```
Comment (spetrosi) [inline guides/doc-Provisioning_Guide/images/ProvisioningUserData-Sequence.txt:13]:
```suggestion
Capsule X --> Satellite: forward Userdata on port 80
```

File: guides/doc-Provisioning_Guide/images/ProvisioningUserData-Sequence.txt [hunk 3]
```diff
@@ -0,0 +1,16 @@
+title VMware Template Provisioning
+participant User
+participant Satellite
+participant VCenter
+participant Capsule X
+
+participant VM
+
+User --> Satellite:Provision VM in network X
+Satellite --> VCenter:Create a VM from template (port 443)
```
Comment (spetrosi) [inline guides/doc-Provisioning_Guide/images/ProvisioningUserData-Sequence.txt:10]:
```suggestion
Satellite --> VCenter:Create a VM from a template on port 443
```

File: guides/doc-Provisioning_Guide/images/ProvisioningUserData-Sequence.txt [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+VCenter --> VM: Clones VM from template
+VM --> Capsule X: Call Back to Sat port 80 via Capsule X 8000
+Capsule X --> Satellite: forward Userdata on 80
+Satellite --> Satellite: redirect to 443
```
Comment (spetrosi) [inline guides/doc-Provisioning_Guide/images/ProvisioningUserData-Sequence.txt:14]:
```suggestion
activate Satellite
Satellite --> Satellite: redirect to 443
deactivate Satellite
```
Add a box to make it clear what redirects where.

File: guides/doc-Provisioning_Guide/images/ProvisioningUserData-Sequence.txt [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+VM -->Capsule X: Registers itself to Satellite
+Capsule X -->Satellite: proxy registration to satellite
```
Comment (spetrosi) [inline guides/doc-Provisioning_Guide/images/ProvisioningUserData-Sequence.txt:16]: What's proxy here? Do you refer to Capsule?

Comment (ckyriakidou) [reply guides/doc-Provisioning_Guide/images/ProvisioningUserData-Sequence.txt:16]: Capsule proxies registration to satellite. That's what I was referring to.

**Comment by ekohl** [comment]
Looks like @spetrosi already took full advantage of the plain text version to leave inline comments. That's a nice side benefit.

File: guides/doc-Provisioning_Guide/images/ProvisioningUserData-Sequence.txt [hunk 6]
```diff
@@ -0,0 +1,16 @@
+title VMware Template Provisioning
+participant User
+participant Satellite
+participant VCenter
```
Comment (melcorr) [inline guides/doc-Provisioning_Guide/images/ProvisioningUserData-Sequence.txt:4]:
```suggestion
participant vCenter
```

File: guides/doc-Provisioning_Guide/images/ProvisioningUserData-Sequence.txt (hunk 5 above)
Comment (spetrosi) [reply guides/doc-Provisioning_Guide/images/ProvisioningUserData-Sequence.txt]: Sorry, I do not get it. I probably lack some context about Capsules. Do you mean smart proxies by chance?

Comment (ckyriakidou) [reply guides/doc-Provisioning_Guide/images/ProvisioningUserData-Sequence.txt]: Capsule forwards registration to satellite. I think that's probably the more understandable way of saying it.

File: guides/doc-Provisioning_Guide/images/ProvisioningUserData-Sequence.txt [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+participant VM
+
+User --> Satellite:Provision VM in network X
+Satellite --> VCenter:Create a VM from a template on port 443
+VCenter --> VM: Clone VM from template
+VM --> Capsule X: Call Back to Sat port 80 via Capsule X 8000
+Capsule X --> Satellite: forward Userdata on port 80
+Satellite --> Satellite: redirect to 443
+VM -->Capsule X: Registers itself to Satellite
+Capsule X -->Satellite: proxy registration to satellite
```
Comment (spetrosi) [inline guides/doc-Provisioning_Guide/images/ProvisioningUserData-Sequence.txt:16]:
```suggestion
Capsule X -->Satellite: Forward VM registration to Satellite
```

File: guides/doc-Provisioning_Guide/images/ProvisioningUserData-Sequence.txt (hunk 5 above)
Comment (spetrosi) [reply guides/doc-Provisioning_Guide/images/ProvisioningUserData-Sequence.txt]: Ah, thank you for clarification. I did not get that "proxy" is a verb here.

File: guides/doc-Provisioning_Guide/images/ProvisioningUserData-Sequence.txt (hunk 4 above)
Comment (spetrosi) [reply guides/doc-Provisioning_Guide/images/ProvisioningUserData-Sequence.txt:14]: If you think this does not work well, we can change the wording to make it more clear what redirects where to `Redirect Userdata to port 443`

File: guides/doc-Provisioning_Guide/images/ProvisioningUserData-Sequence.txt [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+participant Capsule X
+
+participant VM
+
+User --> Satellite:Provision VM in network X
+Satellite --> VCenter:Create a VM from a template on port 443
+VCenter --> VM: Clone VM from template
+VM --> Capsule X: Call Back to Sat port 80 via Capsule X 8000
+Capsule X --> Satellite: forward Userdata on port 80
+Satellite --> Satellite: redirect to 443
```
Comment (spetrosi) [inline guides/doc-Provisioning_Guide/images/ProvisioningUserData-Sequence.txt:14]:
```suggestion
Satellite --> Satellite: Redirect Userdata to port 443
```

File: guides/doc-Provisioning_Guide/images/ProvisioningUserData-Sequence.txt [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+participant vCenter
+participant Capsule X
+
+participant VM
+
+User --> Satellite:Provision VM in network X
+Satellite --> VCenter:Create a VM from a template on port 443
+VCenter --> VM: Clone VM from template
+VM --> Capsule X: Call Back to Sat port 80 via Capsule X 8000
+Capsule X --> Satellite: forward Userdata on port 80
```
Comment (spetrosi) [inline guides/doc-Provisioning_Guide/images/ProvisioningUserData-Sequence.txt:13]:
```suggestion
Capsule X --> Satellite: Forward Userdata on port 80
```
I missed one last capitalization issue.

File: guides/doc-Provisioning_Guide/topics/Virt-VMWare.adoc [hunk 10]
```diff
@@ -258,6 +258,8 @@ You can use the same procedures to set up a VMware compute resource within {Proj
 
 .VMware cloud-init Provisioning Overview
 
+image:ProvisioningUserData-Sequence.png[]
```
Comment (spetrosi) [inline guides/doc-Provisioning_Guide/topics/Virt-VMWare.adoc:261]: Can you please re-upload the diagram with all the latest corrections?

**Pull request merged**